scripts/irods/test/test_prep_genquery_iterator.py

```diff
@@ -127,7 +133,195 @@ def setUp(self):
+                          conditions="META_COLL_ATTR_NAME = 'aa' and COLL_NAME = '{{c}}'".format(c=TestCollection)
+                )
+                d0 = [ i for i in q ]
+                d1 = [ i for i in q.copy(options=Option.NO_DISTINCT) ]
+                callback.msiModAVUMetadata("-C",TestCollection,"rmw", "aa", "bb", "%")
+                b = repr([len(d0),len(d1)])
+                callback.writeLine('stdout',b)
+                callback.msiDeleteUnusedAVUs()
+                '''),
+            "no_distinct_",
+            lambda : eval(output) == [1,2]
+        ), ( #----------------
+            frame_rule('''\
+                import itertools
+                q = Query(callback,
+                          columns=["order_asc(DATA_NAME)"],
+                          conditions="DATA_NAME like '03%' and COLL_NAME = '{{c}}'".format(c=TestCollection)
+                )
+                q2 = q.copy(conditions="DATA_NAME like '05%' and COLL_NAME = '{{c}}'".format(c=TestCollection))
+                chained_queries = itertools.chain(q, q2)
+                compare_this = [x for x in ('%04o'%y for y in range({max_count}+1)) if x[:2] in ('03','05')]
+                callback.writeLine('stdout',repr([x for x in chained_queries] == compare_this and len(compare_this)==128))
+                '''),
+            "iter_chained_queries_",
+            lambda : eval(output) is True
+        ), ( #----------------
+            frame_rule('''\
+                #import pprint # - for DEBUG only
+                import itertools
+                q = Query(callback,
+                          columns=["DATA_NAME"],
+                          conditions="COLL_NAME = '{{c}}'".format(c=TestCollection)
+                )
+                n_results={max_count}+1
+                chunking_results = dict()
+                for chunk_size in set([1,2,3,4,5,6,7,8,10,16,32,50,64,100,128,200,256,400,510,512,513,
+                 n_results-2,n_results-1,n_results,n_results+1,n_results+2]):
+                    q_iter = iter(q.copy())
+                    L=[]
+                    while True:
+                        chunk = [row for row in itertools.islice(q_iter, 0, chunk_size)]
+                        if not chunk: break
+                        L.append(chunk)
+                    chunking_results[chunk_size] = L
+                check_Lresult = lambda chunkSize,L:\
+                  [int(x[0],8) for x in L] == list(range(0,n_results,chunkSize)) and \
+                  (0==len(L) or 1+((n_results-1)%chunkSize) == len(L[-1]))
+                #callback.writeLine('serverLog',pprint.pformat(chunking_results)) # - for DEBUG only
+                callback.writeLine('stdout',repr(list(check_Lresult(k,chunking_results[k]) for k in chunking_results)))
+                '''),
+            "chunked_queries_via_islice_",
+            lambda : list(filter(lambda result:result is False, eval(output))) == []
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of possibly insecure function - consider using safer ast.literal_eval.](https://app.codacy.com/manual/trel/irods/pullRequest?prid=5016322)